cuda: sample build fails when enabling CUDA

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( 8d7ccd2 or later )
- Operating System / Platform => Jetson TX1, TX2, Arm 64bit
- Compiler => GCC 5.4.0

##### Detailed description

- The build of sample program fails when enabling CUDA
```
In file included from /home/ubuntu/opencv-fork/samples/cpp/stitching_detailed.cpp:52:0:
/home/ubuntu/opencv-fork/modules/stitching/include/opencv2/stitching/detail/blenders.hpp:153:18: error: field ‘gpu_dst_mask_’ has incomplete type ‘cv::cuda::GpuMat’
     cuda::GpuMat gpu_dst_mask_;
                  ^
In file included from /home/ubuntu/opencv-fork/modules/core/include/opencv2/core.hpp:54:0,
                 from /home/ubuntu/opencv-fork/modules/core/include/opencv2/core/utility.hpp:56,
                 from /home/ubuntu/opencv-fork/samples/cpp/stitching_detailed.cpp:48:
/home/ubuntu/opencv-fork/modules/core/include/opencv2/core/base.hpp:726:11: note: forward declaration of ‘class cv::cuda::GpuMat’
     class CV_EXPORTS GpuMat;
           ^
```
- I'll send a PR later

##### Steps to reproduce
```
mkdir build
cd build
cmake -DBUILD_EXAMPLES:BOOL=ON -DWITH_CUDA=ON .. 
make -j 4 all
```

--- a/modules/stitching/include/opencv2/stitching/detail/blenders.hpp
+++ b/modules/stitching/include/opencv2/stitching/detail/blenders.hpp
@@ -48,6 +48,7 @@
 #endif
 
 #include "opencv2/core.hpp"
+#include "opencv2/core/cuda.hpp"
 
 namespace cv {
 namespace detail {

